### DISCUSSION TITLE
`enabled` truthy values

### DISCUSSION THREAD

**Discussion by timothedelion** [TanStack/query#4748] [General]
`enabled` is considered falsy only if it is strictly equal to `false`.

So if you pass `undefined` or `""` to `enabled`, it will be considered truthy and the query will be triggered. Which is not what I expect intuitively.

What do you think about casting the input value as a `Boolean` ? Am I missing a part of the picture ?

**Comment by TkDodo** [comment]
we've had this in v2 and moved away from this in v3. We now:
- have TypeScript that tells you what to do
- throw a runtime error if you try to pass in something that's not a boolean:

https://github.com/TanStack/query/blob/af2e27ef377fd8c0e99cc9251f18017267509814/packages/query-core/src/queryObserver.ts#L176-L181

I'm not really sure why we don't allow non-booleans. I think being explicit about your intentions doesn't hurt though :)
